src/models/sample.js

```diff
@@ -6,6 +6,36 @@ import { parseISODate, DATE_FORMAT_DISPLAY } from '../shared/util';
 
 class Sample {}
 
+// To clean up after mapping single field to object for backend
+Sample.prepareForSubmit = tmpData => {
```

Comment:
Can you rewrite this to a more FP approach. You don't need to do it now but there's some hints in the AnalysisAdd component on how you can solve it.